python/ray/tests/test_advanced_2.py

```diff
@@ -599,11 +599,17 @@ def test():
         while resources and retry_count < MAX_RETRY_ATTEMPTS:
             time.sleep(0.1)
             resources = ray.available_resources()
+            del resources["memory"]
```

Comment:
Does it do anything? Isn't `available_resources` just calling raylet and get the result? This will just update the in-memory dictionary, but this dict will be renewed in the next call.

Reply:
`available_resources` return results containing keys including `memory`/`cpu`/`node:xxxx` etc. in this case `cpu` and `gpu` have already been decreased to zero and not included in the returned dict. So we need delete other unrelated keys then can use nullable check of resource (dict) every time we got the return value.
The original case just delete unrelated keys in the first value getting but not in the retry loop, this PR just add the missing deleting. @rkooo567

Reply:
If we do not delete these keys then the `resources`  will never be empty in the retries, which makes the retry loop non-sense.

Reply:
Ah, that makes a lot of sense. Can you put this logic into a function and just run a function in this case? (since it looks like it is the copy of the logic above).

Reply:
done :) @rkooo567 
The failed cases are not related to this change.